### ISSUE TITLE
[Feature]: Change checkboxes in issue template 

### ISSUE THREAD

**Issue by rly** [hdmf-dev/hdmf#1003]
### What would you like to see added to HDMF?

Checkboxes in GitHub issues are [intended to represent tasks that need to be done to resolve the issue](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/about-task-lists), but our issue checkboxes effectively act to ensure the issue submitter agrees to our terms for submitting an issue. This results in a confusing GitHub interface and options that do not make sense. I know at least one other person who was confused by this ("The issue says all of the tasks are done!")

<img width="906" alt="image" src="https://github.com/hdmf-dev/hdmf/assets/310197/ae28cf1d-2a74-4aac-b3f9-e3fa9938465b">

<img width="881" alt="Screenshot 2023-11-27 at 12 25 48 AM" src="https://github.com/hdmf-dev/hdmf/assets/310197/5dff6e13-87a1-48d7-9c64-7be5a9256018">

Also related: it would be nice to simplify the new issue form a bit. For new features, the second box "Is your feature request related to a problem?" could be folded into the first box "What would you like to see added to HDMF?". In practice I enter everything in the first box and leave the second empty.

### What solution would you like?

Change issue checkboxes to text.

PR checkboxes can still be useful because they represent tasks that should be done before merging the PR. However, even these can be trimmed down / simplified.

### Do you have any interest in helping implement the feature?

Yes.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/hdmf-dev/hdmf/blob/dev/.github/CODE_OF_CONDUCT.md)
- [X] Have you checked the [Contributing](https://github.com/hdmf-dev/hdmf/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured this change was not already [requested](https://github.com/hdmf-dev/hdmf/issues)?
